gensim/test/test_ldamodel_distributed.py

```diff
@@ -0,0 +1,527 @@
+        self.assertFalse(np.allclose(model1.eta, modelauto.eta))
+
+    def testEta(self):
+        kwargs = dict(
+            id2word=dictionary,
+            num_topics=2,
+            eta=None
+        )
+        num_terms = len(dictionary)
+        expected_shape = (num_terms,)
+
+        # should not raise anything
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.5] * num_terms))
+
+        kwargs['eta'] = 'symmetric'
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.5] * num_terms))
+
+        kwargs['eta'] = 0.3
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.3] * num_terms))
+
+        kwargs['eta'] = 3
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([3] * num_terms))
+
+        kwargs['eta'] = [0.3] * num_terms
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.3] * num_terms))
+
+        kwargs['eta'] = np.array([0.3] * num_terms)
+        model = self.class_(**kwargs)
+        self.assertEqual(model.eta.shape, expected_shape)
+        assert_allclose(model.eta, np.array([0.3] * num_terms))
+
+        # should be ok with num_topics x num_terms
+        testeta = np.array([[0.5] * len(dictionary)] * 2)
+        kwargs['eta'] = testeta
+        self.class_(**kwargs)
+
+        # all should raise an exception for being wrong shape
+        kwargs['eta'] = testeta.reshape(tuple(reversed(testeta.shape)))
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['eta'] = [0.3]
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['eta'] = [0.3] * (num_terms + 1)
+        self.assertRaises(AssertionError, self.class_, **kwargs)
+
+        kwargs['eta'] = "gensim is cool"
+        self.assertRaises(ValueError, self.class_, **kwargs)
+
+        kwargs['eta'] = "asymmetric"
+        self.assertRaises(ValueError, self.class_, **kwargs)
+
+    def testTopTopics(self):
+        top_topics = self.model.top_topics(self.corpus)
+
+        for topic, score in top_topics:
+            self.assertTrue(isinstance(topic, list))
+            self.assertTrue(isinstance(score, float))
+
+            for v, k in topic:
+                self.assertTrue(isinstance(k, six.string_types))
+                self.assertTrue(np.issubdtype(v, np.floating))
+
+    def testGetTopicTerms(self):
+        topic_terms = self.model.get_topic_terms(1)
+
+        for k, v in topic_terms:
+            self.assertTrue(isinstance(k, numbers.Integral))
+            self.assertTrue(np.issubdtype(v, np.floating))
+
+    def testGetDocumentTopics(self):
```

Comment:
There's a lot of logic in this test method. That's not a good thing, because logic is prone to bugs, and you risk introducing bugs into actual test code (which itself does **not** get tested by anything).

For this reason, test code should be as simple as possible.

Furthermore, this method is far too long for a single test. There are many places where the test can fail. This is bad from a maintainer's (mine) perspective, because when a test fails, I'll have to work out what exactly it's doing and why it's failing. There's a lot to work out here.

Some suggestions:

- Break this test up into multiple smaller tests, if possible
- Extract common funcitonality into pure functions, and reuse them. For instance, much of your code here and in other tests performs type checking, e.g.
  1. Is the variable a list?
  2. Is each element of the list a tuple of some type?

You should also avoid loops in test code, if possible. Instead of iterating through a dictionary and checking items one by one, create a dictionary of expected values, and compare that to actual values in one assertion.